crates/iceberg/src/arrow/schema.rs

```diff
@@ -385,25 +389,236 @@ impl ArrowSchemaVisitor for ArrowSchemaConverter {
+            }
+            crate::spec::PrimitiveType::Double => {
+                Ok(ArrowSchemaOrFieldOrType::Type(DataType::Float64))
+            }
+            crate::spec::PrimitiveType::Decimal { precision, scale } => {
+                let (precision, scale) = {
+                    let precision: u8 = precision.to_owned().try_into().map_err(|err| {
+                        Error::new(
+                            crate::ErrorKind::DataInvalid,
+                            "incompatible precision for decimal type convert",
+                        )
+                        .with_source(err)
+                    })?;
+                    let scale = scale.to_owned().try_into().map_err(|err| {
+                        Error::new(
+                            crate::ErrorKind::DataInvalid,
+                            "incompatible scale for decimal type convert",
+                        )
+                        .with_source(err)
+                    })?;
+                    (precision, scale)
+                };
+                validate_decimal_precision_and_scale::<Decimal128Type>(precision, scale).map_err(
+                    |err| {
+                        Error::new(
+                            crate::ErrorKind::DataInvalid,
+                            "incompatible precision and scale for decimal type convert",
+                        )
+                        .with_source(err)
+                    },
+                )?;
+                Ok(ArrowSchemaOrFieldOrType::Type(DataType::Decimal128(
+                    precision, scale,
+                )))
+            }
+            crate::spec::PrimitiveType::Date => {
+                Ok(ArrowSchemaOrFieldOrType::Type(DataType::Date32))
+            }
+            crate::spec::PrimitiveType::Time => Ok(ArrowSchemaOrFieldOrType::Type(
+                DataType::Time32(TimeUnit::Microsecond),
+            )),
+            crate::spec::PrimitiveType::Timestamp => Ok(ArrowSchemaOrFieldOrType::Type(
+                DataType::Timestamp(TimeUnit::Microsecond, None),
+            )),
+            crate::spec::PrimitiveType::Timestamptz => Ok(ArrowSchemaOrFieldOrType::Type(
+                // Timestampz always stored as UTC
+                DataType::Timestamp(TimeUnit::Microsecond, Some("+00:00".into())),
+            )),
+            crate::spec::PrimitiveType::String => {
+                Ok(ArrowSchemaOrFieldOrType::Type(DataType::Utf8))
+            }
+            crate::spec::PrimitiveType::Uuid => Ok(ArrowSchemaOrFieldOrType::Type(
+                DataType::FixedSizeBinary(16),
+            )),
+            crate::spec::PrimitiveType::Fixed(len) => Ok(ArrowSchemaOrFieldOrType::Type(
+                len.to_i32()
+                    .map(DataType::FixedSizeBinary)
+                    .unwrap_or(DataType::LargeBinary),
+            )),
+            crate::spec::PrimitiveType::Binary => {
+                Ok(ArrowSchemaOrFieldOrType::Type(DataType::LargeBinary))
```

Comment:
Why a large binary over here, and not a large-string for example?

Reply:
I think Binary is a byte array and we should not change it as string?🤔

Reply:
+1 for large binary. I think there is a difference between binary and string? Also again, there is no string type in arrow.